File: BK_BSMTV/csharp/UserService.cs
using System;
using System.Collections.Generic;

namespace MiProyectoAdministracion
{
    public class UserService
    {
        private List<User> users;

        public UserService()
        {
            users = new List<User>();
        }

        public void AddUser(User user)
        {
            users.Add(user);
        }

        public void RemoveUser(string userId)
        {
            users.RemoveAll(u => u.Id == userId);
        }

        public User GetUser(string userId)
        {
            return users.Find(u => u.Id == userId);
        }

        public List<User> GetAllUsers()
        {
            return users;
        }

        public void UpdateUser(User updatedUser)
        {
            var index = users.FindIndex(u => u.Id == updatedUser.Id);
            if (index != -1)
            {
                users[index] = updatedUser;
            }
        }
    }
}

File: public/BK_BSMTV/frontend/angular-app/src/app/user/user.service.ts
export class UserService {
  private users: any[] = [];

  constructor() {}

  addUser(user: any): void {
    this.users.push(user);
  }

  getUsers(): any[] {
    return this.users;
  }

  updateUser(index: number, updatedUser: any): void {
    if (index >= 0 && index < this.users.length) {
      this.users[index] = updatedUser;
    }
  }

  deleteUser(index: number): void {
    if (index >= 0 && index < this.users.length) {
      this.users.splice(index, 1);
    }
  }
}